src/store/swapOrderSlice.ts

```diff
@@ -0,0 +1,197 @@
+import type { listenerMiddlewareInstance } from '@/store'
+import { createSelector, createSlice } from '@reduxjs/toolkit'
+import type { OrderStatuses } from '@safe-global/safe-gateway-typescript-sdk'
+import type { RootState } from '@/store'
+import { isSwapTxInfo, isTransactionListItem } from '@/utils/transaction-guards'
+import { txHistorySlice } from '@/store/txHistorySlice'
+import { showNotification } from '@/store/notificationsSlice'
+import { selectSafeInfo } from '@/store/safeInfoSlice'
+
+type AllStatuses = OrderStatuses | 'created'
+type Order = {
+  orderUid: string
```

Comment:
With this structure we have `orderUid` twice. Is it needed or can we remove the object attribute?

Reply:
It's twice, but get used to that. It's a normal pattern in redux and it simplifies working with the object. I know you have it in the key, but then you would always have to do Object.keys() to get it... It's very annoying. The duplication here is fine.

Reply:
Agree that its more convenient and we are not optimizing for space so lets follow this pattern in the future.